query-engine/connectors/sql-query-connector/src/query_builder/write.rs

```diff
@@ -161,10 +161,14 @@ pub fn update_many(
         });
 
     let query = query.append_trace(&Span::current()).add_trace_id(trace_id);
-    let columns: Vec<_> = ModelProjection::from(model.primary_identifier()).as_columns().collect();
-    let result: Vec<Query> = super::chunked_conditions(&columns, ids, |conditions| {
-        query.clone().so_that(conditions.and(filter_condition.clone()))
-    });
+    let result: Vec<Query> = if ids.is_empty() {
+        Vec::from([query.so_that(filter_condition).into()])
```

Comment:
If I understand this right, the ids being empty is the way we signal that the updates doesn’t need the `IN` clause. I’m not familiar enough with this part of the code to say, but would here be the right place for a comment? Also, do we ever get here if we did do a select, but it returned no IDs?

Reply:
Yes correct. It also means we don't need to chunk the update because we are not going to have a huge list of parameters. 

This line here https://github.com/prisma/prisma-engines/pull/3197/files#diff-f25c0ba73dfc90e47e302e5f0cecd3f4f03dac4a0f19535bd94aa56aa9fb932cR349
Checks that if the was no ids returned from the select then it never goes to the update.